comp/core/gui/guiimpl/auth.go

```diff
@@ -0,0 +1,135 @@
+	}
+
+	// Check token version
+	if parts[0] != tokenVersion {
+		return fmt.Errorf("token version mismatch: got %s, expected %s", parts[0], tokenVersion)
+	}
+
+	// Decode the payload from base64
+	payloadBytes, err := base64.StdEncoding.DecodeString(parts[1])
+	if err != nil {
+		return fmt.Errorf("failed to decode payload: %w", err)
+	}
+
+	// Ensure the payload contains enough bytes for issued and expiration times
+	if len(payloadBytes) < 16 {
+		return fmt.Errorf("invalid payload")
+	}
+
+	// Extract the issued and expiration times from the payload
+	issuedTime := int64(binary.LittleEndian.Uint64(payloadBytes[:8]))
+	expirationTime := int64(binary.LittleEndian.Uint64(payloadBytes[8:]))
+
+	// Decode the HMAC sum from base64
+	hmacSum, err := base64.StdEncoding.DecodeString(parts[2])
+	if err != nil {
+		return fmt.Errorf("failed to decode HMAC sum: %w", err)
+	}
+
+	// Calculate the expected HMAC sum
+	mac := hmac.New(sha256.New, a.signingKey)
+	mac.Write(payloadBytes)
+	expectedHmacSum := mac.Sum(nil)
+
+	// Check if the HMAC sum matches the expected HMAC sum
+	if !hmac.Equal(hmacSum, expectedHmacSum) {
+		return fmt.Errorf("invalid token signature")
+	}
+
+	now := time.Now()
+
+	// Check if the current time is before the issued time
+	if now.Before(time.Unix(issuedTime, 0)) {
+		return fmt.Errorf("token is invalid")
+	}
+
+	// special case: ignore expirationTime if duration is equal to 0
+	// Check if the current time is after the expiration time
+	if expirationTime != issuedTime && now.After(time.Unix(expirationTime, 0)) {
+		return fmt.Errorf("token is expired")
+	}
+
+	if a.duration != 0 && now.After(time.Unix(issuedTime, 0).Add(a.duration)) {
+		return fmt.Errorf("token is expired")
+	}
+
+	return nil
+}
+
+// hmacToken generates an HMAC token using the provided key, issued time, and expiration time.
+// The token is generated by converting the issued time and expiration time to base64 unixTime format,
+// creating an HMAC sum using the key and payload, and combining the payload and HMAC sum with a "." separator.
+// The generated token is returned as a string.
+//
+// Token representation:
+// +----------------+----------------+--------------------------------+----------------+----------------------------------+
+// | Token Version  | "." Separator  | Base64 Encoded Payload         | "." Separator  | Base64 Encoded HMAC Sum          |
+// +----------------+----------------+--------------------------------+----------------+----------------------------------+
+// | ~3 characters  | 1 character    | ~24 characters                 | 1 character    | ~44 characters                   |
+// +----------------+----------------+--------------------------------+----------------+----------------------------------+
+// with payload
+// +----------------+----------------+
+// | Issued Time    | Expiration Time|
+// +----------------+----------------+
+// | 8 bytes        | 8 bytes        |
+// +----------------+----------------+
```

Comment:
Should this be on top of `Token representation`? I'm guessing this is explaining the `hmacToken` arguments, right? Alos, it is missing the `token` information?

Reply:
It is explaining how is composed the `Base64 Encoded Payload` from the table above.